### PR TITLE
[4.0] Fix PHP 8.1 deprecation - strtolower null param

### PR DESCRIPTION
**Pull request by PhilETaylor**
### Summary of Changes

Set sane default if the configuration.php has no value for `dbtype`, for example when running unit tests.

### Testing Instructions

on a php 8.1 command line run 

`libraries/vendor/bin/phpunit tests/Unit/Libraries/Cms/Date/DateTest.php`

### Actual result BEFORE applying this Pull Request

```
libraries/vendor/bin/phpunit tests/Unit/Libraries/Cms/Date/DateTest.php

PHPUnit 8.5.21 by Sebastian Bergmann and contributors.

...PHP Deprecated:  strtolower(): Passing null to parameter #1 ($string) of type string is deprecated in /Users/phil/Sites/joomla-cms/libraries/src/Service/Provider/Database.php on line 60
PHP Deprecated:  strtolower(): Passing null to parameter #1 ($string) of type string is deprecated in /Users/phil/Sites/joomla-cms/libraries/src/Service/Provider/Database.php on line 65
PHP Deprecated:  strtolower(): Passing null to parameter #1 ($string) of type string is deprecated in /Users/phil/Sites/joomla-cms/libraries/src/Service/Provider/Database.php on line 78
PHP Deprecated:  strtolower(): Passing null to parameter #1 ($string) of type string is deprecated in /Users/phil/Sites/joomla-cms/libraries/src/Service/Provider/Database.php on line 111
PHP Deprecated:  strtolower(): Passing null to parameter #1 ($string) of type string is deprecated in /Users/phil/Sites/joomla-cms/libraries/src/Service/Provider/Database.php on line 116
.
Deprecated: strtolower(): Passing null to parameter #1 ($string) of type string is deprecated in /Users/phil/Sites/joomla-cms/libraries/src/Service/Provider/Database.php on line 60

Deprecated: strtolower(): Passing null to parameter #1 ($string) of type string is deprecated in /Users/phil/Sites/joomla-cms/libraries/src/Service/Provider/Database.php on line 65

Deprecated: strtolower(): Passing null to parameter #1 ($string) of type string is deprecated in /Users/phil/Sites/joomla-cms/libraries/src/Service/Provider/Database.php on line 78

Deprecated: strtolower(): Passing null to parameter #1 ($string) of type string is deprecated in /Users/phil/Sites/joomla-cms/libraries/src/Service/Provider/Database.php on line 111

Deprecated: strtolower(): Passing null to parameter #1 ($string) of type string is deprecated in /Users/phil/Sites/joomla-cms/libraries/src/Service/Provider/Database.php on line 116
......                                                        10 / 10 (100%)

Time: 52 ms, Memory: 10.00 MB

OK (10 tests, 32 assertions)
```

### Expected result AFTER applying this Pull Request

```
libraries/vendor/bin/phpunit tests/Unit/Libraries/Cms/Date/DateTest.php


PHPUnit 8.5.21 by Sebastian Bergmann and contributors.

..........                                                        10 / 10 (100%)

Time: 57 ms, Memory: 10.00 MB

OK (10 tests, 32 assertions)
```

### CODE REVIEW

File: libraries/src/Service/Provider/Database.php [hunk 1]
```diff
@@ -42,9 +42,19 @@ public function register(Container $container)
...
+					$dbtype = $conf->get('dbtype', null);
+
+					// Ensure we have a db type, if we don't then set a sane, assumed, default.
+					if (null === $dbtype)
+					{
+						$dbtype = 'mysqli';
+					}
+
+					// Normalise.
+					$dbtype = strtolower($dbtype);
```
Comment (Quy) [inline libraries/src/Service/Provider/Database.php]:
```suggestion
					// Ensure we have a db type, if we don't then set a sane, assumed, default.
					$dbtype = $dbtype === null ? 'mysqli' : strtolower($dbtype);
```
Simplify version?

Comment (PhilETaylor) [reply libraries/src/Service/Provider/Database.php]: Thanks.

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on e5aa48fb9ac609e977dee61fc3fa286778befae7<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36397">issues.joomla.org/tracker/joomla-cms/36397</a>.</sub>

**Comment by Quy** [comment]
Please fix conflicts. Thanks.

**Comment by PhilETaylor** [comment]
hmm on it thanks

**Comment by PhilETaylor** [comment]
Done. Seems someone else did the same kind of thing, but this PR also cleans up the file and reduces the amount of code - so still has value.

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 12b40e2b7ff63b10a08eae36835cd29aa429438c<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36397">issues.joomla.org/tracker/joomla-cms/36397</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36397">issues.joomla.org/tracker/joomla-cms/36397</a>.</sub>

**Pull request closed without merging**